### PR TITLE
Update loadsave.cpp

### PR DESCRIPTION
**Pull request by guoxuedong**
add scale param when convert from CV_16U/CV_16S/CV_32S/CV_32F/CV_64F to CV_8U.

### CODE REVIEW

**Comment by guoxuedong** [comment]
imwrite  returns an all white image when read a CV_16U type image by imread.
I found convert from CV_16U to CV_8U simply invokes saturate_cast<>, and made pixel value all 255.
And I think it is better to add scale param when invoke convertTo in imwrite and imencode.

File: modules/imgcodecs/src/loadsave.cpp [hunk 1]
```diff
@@ -518,6 +518,29 @@ bool imreadmulti(const String& filename, std::vector<Mat>& mats, int flags)
     return imreadmulti_(filename, flags, mats);
 }
 
+static void
+convertToCV8U(const Mat& src, const Mat& dst, int rtype)
+{
+    switch (rtype)
```
Comment (vpisarev) [inline modules/imgcodecs/src/loadsave.cpp]: why not simply write `src.depth()` here instead of `rtype`? We can eliminate `rtype` parameter

**Comment by vpisarev** [comment]
@guoxuedong, thanks for the patch! Please, see the comment above

**Comment by alalek** [comment]
Related: #7956 #7762([comment](https://github.com/opencv/opencv/issues/7762#issuecomment-270105236))

File: modules/imgcodecs/src/loadsave.cpp (hunk 1 above)
Comment (guoxuedong) [inline modules/imgcodecs/src/loadsave.cpp]: Thanks for you review! 
You are right and I fixed it. Please review this change.

**Comment by guoxuedong** [comment]
@alalek , Thanks for your comment!
I read these issue and I don't know should I do something ?

**Comment by guoxuedong** [comment]
@vpisarev ,Thanks for you review!
You are right and I fixed it. Please review this change.

**Comment by vpisarev** [comment]
@guoxuedong, we discussed this patch a bit more with @alalek. It looks like the patch may create some problems. It may look convenient for users, but it may create some confusion. If the selected format (derived from the file extension) supports 16-bit or 32-bit format, then the corresponding data can be saved w/o any implicit transformations. If not, then the data may be converted to 8-bit, as you suggested in the patch. But it's not obvious, which particular conversion user needs. Whether 16-bit data should be scaled by dividing it by 256? Or user actually stored 12-bit data and want to divide it just by 16? Same with floating-point. Should 0..1 range be used? or 0..255? or 0..256?

Given that there is Mat::convertTo() method, that can be used to convert data to 8-bit explicitly, and it's just one line of code, we decided that user should do this conversion explicitly to avoid confusion.

**Pull request closed without merging**